esmvaltool/preprocessor/_derive.py

```diff
@@ -80,6 +90,32 @@ def derive(cubes, variable):
     return cube
 
 
+def calc_lwcre(cubes):
+    """Compute longwave cloud radiative effect from all-sky and clear-sky flux.
+
+    Arguments
+    ----
+        cubes: cubelist containing rlut (toa_outgoing_longwave_flux) and rlutcs
+               (toa_outgoing_longwave_flux_assuming_clear_sky).
+
+    Returns
+    -------
+        Cube containing longwave cloud radiative effect.
+
+    """
+    rlut_cube = cubes.extract_strict(
+        Constraint(name='toa_outgoing_longwave_flux'))
+    rlutcs_cube = cubes.extract_strict(
+        Constraint(name='toa_outgoing_longwave_flux_assuming_clear_sky'))
+
+    assert rlut_cube.coord_dims('time') and rlutcs_cube.coord_dims('time'), \
```

Comment:
Why this check? Is it relevant to this derivation? If the cube has no time coordinate, preprocessing will probably already fail at the `extract_time` function.